test/e2e/latency.go

```diff
@@ -0,0 +1,220 @@
+	"sort"
+	"strconv"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client/cache"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/controller/framework"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/watch"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// This test suite can take a long time to run, so by default it is added to
+// the ginkgo.skip list (see driver.go).
+// To run this suite you must explicitly ask for it by setting the
+// -t/--test flag or ginkgo.focus flag.
+var _ = Describe("Density", func() {
+	var c *client.Client
+	var minionCount int
+	var RCName string
+	var additionalRCName string
+	var ns string
+	var uuid string
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		expectNoError(err)
+		minions, err := c.Nodes().List(labels.Everything(), fields.Everything())
+		expectNoError(err)
+		minionCount = len(minions.Items)
+		Expect(minionCount).NotTo(BeZero())
+		nsForTesting, err := createTestingNS("density", c)
+		ns = nsForTesting.Name
+		expectNoError(err)
+		uuid = string(util.NewUUID())
+	})
+
+	AfterEach(func() {
+		// Remove any remaining pods from this test if the
+		// replication controller still exists and the replica count
+		// isn't 0.  This means the controller wasn't cleaned up
+		// during the test so clean it up here
+		rc, err := c.ReplicationControllers(ns).Get(RCName)
+		if err == nil && rc.Spec.Replicas != 0 {
+			By("Cleaning up the replication controller")
+			err := DeleteRC(c, ns, RCName)
+			expectNoError(err)
+		}
+
+		rc, err = c.ReplicationControllers(ns).Get(additionalRCName)
+		if err == nil && rc.Spec.Replicas != 0 {
+			By("Cleaning up the replication controller")
+			err := DeleteRC(c, ns, additionalRCName)
+			expectNoError(err)
+		}
+
+		By(fmt.Sprintf("Destroying namespace for this suite %v", ns))
+		if err := c.Namespaces().Delete(ns); err != nil {
+			Failf("Couldn't delete ns %s", err)
+		}
+	})
+
+	// Tests with "Skipped" substring in their name will be skipped when running
+	// e2e test suite without --ginkgo.focus & --ginkgo.skip flags.
+	type Latency struct {
+		skip          bool
+		podsPerMinion int
+		/* Controls how often the apiserver is polled for pods */
+		interval int
+	}
+
+	densityTests := []Latency{
+		// This test should always run, even if larger densities are skipped.
+		{podsPerMinion: 3, skip: false, interval: 10},
+		{podsPerMinion: 30, skip: false, interval: 10},
+		// More than 30 pods per node is outside our v1.0 goals.
+		// We might want to enable those tests in the future.
+		{podsPerMinion: 50, skip: true, interval: 10},
+		{podsPerMinion: 100, skip: true, interval: 1},
```

Comment:
I see, well upto you, but I'd like to avoid forking to avoid duplication. Maintaining 2 will be a headache. Maybe you can call the same function with 2 densityTest configs, and then spin up your latency rc at the end of the latency test? 

Our normal e2e tests only run on a 2 node cluster, the e2e-scalability tests run on a big cluster but most of the times no one cares if they take 8 or 10 minutes (currently they're around 8).